src/SuperstateToken.sol

```diff
@@ -383,6 +421,141 @@ abstract contract SuperstateToken is
+    }
+
+    function calculateFee(uint256 amount, uint256 subscriptionFee) public pure returns (uint256) {
+        return (amount * subscriptionFee) / FEE_DENOMINATOR;
+    }
+
+    /**
+     * @notice The ```calculateSuperstateTokenOut``` function calculates the total amount of Superstate tokens you'll receive for the inAmount of stablecoin. Treats all stablecoins as if they are always worth a dollar.
+     * @param inAmount The amount of the stablecoin in
+     * @param stablecoin The address of the stablecoin to calculate with
+     * @return superstateTokenOutAmount The amount of Superstate tokens received for inAmount of stablecoin
+     * @return stablecoinInAmountAfterFee The amount of the stablecoin in after any fees
+     * @return feeOnStablecoinInAmount The amount of the stablecoin taken in fees
+     */
+    function calculateSuperstateTokenOut(uint256 inAmount, address stablecoin) public view returns (uint256 superstateTokenOutAmount, uint256 stablecoinInAmountAfterFee, uint256 feeOnStablecoinInAmount) {
+        StablecoinConfig memory config = supportedStablecoins[stablecoin];
+        if (config.out == address(0)) revert StablecoinNotSupported();
+
+        feeOnStablecoinInAmount = calculateFee({amount: inAmount, subscriptionFee: config.fee});
+        stablecoinInAmountAfterFee = inAmount - feeOnStablecoinInAmount;
+
+        (bool isBadData,, uint256 usdPerSuperstateTokenChainlinkRaw) = _getChainlinkPrice();
+        if (isBadData) revert BadChainlinkData();
+
+        uint256 stablecoinDecimals = IERC20Metadata(stablecoin).decimals();
+        uint256 stablecoinPrecision = 10 ** stablecoinDecimals;
+        uint256 chainlinkFeedPrecision = 10 ** AggregatorV3Interface(superstateOracle).decimals();
+
+        // converts from a USD amount to a SUPERSTATE_TOKEN amount
+        superstateTokenOutAmount = (stablecoinInAmountAfterFee * chainlinkFeedPrecision * SUPERSTATE_TOKEN_PRECISION)
+            / (usdPerSuperstateTokenChainlinkRaw * stablecoinPrecision);
+    }
+
+    /**
+     * @notice The ```subscribe``` function takes in stablecoins and mints SuperstateToken in the proper amount for the msg.sender depending on the current Net Asset Value per Share.
+     * @param inAmount The amount of the stablecoin in
+     * @param stablecoin The address of the stablecoin to calculate with
+     */
+    function subscribe(uint256 inAmount, address stablecoin) external {
+        if (inAmount == 0) revert BadArgs();
+        _requireNotPaused();
+        _requireNotAccountingPaused();
+
+        StablecoinConfig memory config = supportedStablecoins[stablecoin];
+        if (config.out == address(0)) revert StablecoinNotSupported();
+
+        (uint256 superstateTokenOutAmount, uint256 stablecoinInAmountAfterFee,) = calculateSuperstateTokenOut({inAmount: inAmount, stablecoin: stablecoin});
+
+        // TODO: revert if superstateTokenOutAmount below a certain amount? like 0?
```

Comment:
below 0 should be impossible, right. maybe we disallow dust (does it even matter)?

Reply:
Yeah I mean should we revert if 0, or a slightly higher number?

Reply:
I'd say to revert if out is 0, but there could be situations in the future where someone actually wants a dust amount